app_flutter/lib/service/cocoon.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show kReleaseMode;
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import 'appengine_cocoon.dart';
+import 'fake_cocoon.dart';
+
+/// Service class for interacting with flutter/flutter build data.
+///
+/// This service exists as a common interface for getting build data from a data source.
+abstract class CocoonService {
+  /// Creates a new [CocoonService] based on if the Flutter app is in production.
+  ///
+  /// Production uses the Cocoon backend running on AppEngine.
+  /// Otherwise, it uses fake data populated from a fake service.
+  factory CocoonService() {
+    if (kReleaseMode) {
+      return AppEngineCocoonService();
+    }
+
+    // TODO(chillers): LocalCocoonService. https://github.com/flutter/cocoon/issues/442
+
+    return FakeCocoonService();
+  }
+
+  /// Gets build information from the last 200 commits.
+  ///
+  /// TODO(chillers): Make configurable to get range of commits
+  Future<List<CommitStatus>> getStats();
```

Comment:
A general nit on API design - use `get` as a verb of last resort. This also downloads a commit status but as named as returning stats - these should probably match

```suggestion
  Future<List<CommitStatus>> fetchCommitStatuses();
```

fetch is nearly as generic as get, so that isn't a huge improvement either

Reply:
Thank you!